trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1101,6 +1105,83 @@ private Response getTree(UUID expUUID, String outputId, QueryParameters queryPar
         }
     }
 
+    /**
+     * Query the provider for a report. Based on the given output, this
+     * endpoint will return a specific report, which can be an image, a
+     * textual data, a HTML, etc.
+     *
+     * @param expUUID
+     *            desired experiment UUID
+     * @param outputId
+     *            Output ID for the data provider to query
+     * @return {@link Response} with the corresponding report data
+     */
+    @GET
+    @Path("/report/{outputId}")
+    @Tag(name = RPT)
+    @Produces({ MediaType.APPLICATION_OCTET_STREAM, MediaType.APPLICATION_JSON })
+    @Operation(summary = "API to get an image", responses = {
+            @ApiResponse(responseCode = "200", description = "Returns the image file", content = @Content(mediaType = "application/octet-stream")),
+            @ApiResponse(responseCode = "400", description = MISSING_PARAMETERS, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = PROVIDER_NOT_FOUND, content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "500", description = "Error reading the image file", content = @Content(schema = @Schema(implementation = String.class)))
+    })
+    public Response getReport(
+            @Parameter(description = EXP_UUID) @PathParam("expUUID") UUID expUUID,
+            @Parameter(description = OUTPUT_ID) @PathParam("outputId") String outputId) {
+
+        if (outputId == null) {
+            return Response.status(Status.BAD_REQUEST).entity(MISSING_OUTPUTID).build();
+        }
+
+        TmfExperiment experiment = ExperimentManagerService.getExperimentByUUID(expUUID);
+        if (experiment == null) {
+            return Response.status(Status.NOT_FOUND).entity(NO_SUCH_TRACE).build();
+        }
+
+        IDataProviderDescriptor descriptor = getDescriptor(experiment, outputId);
+        if (descriptor == null || descriptor.getType() != ProviderType.REPORT) {
+            return Response.status(Status.NOT_FOUND).entity(NO_SUCH_PROVIDER).build();
+        }
+
+        ProviderType reportType = descriptor.getSubType();
+        if (reportType == null) {
+            return Response.status(Status.NOT_FOUND).entity(NO_SUCH_PROVIDER).build();
+        }
+
+        ITmfConfiguration config = descriptor.getConfiguration();
```

Comment:
We have an issue here. The report is a customization for a that is specific for the Trace Compass server and won't exists in other server implementation. This endpoint needs to be independent from it. 

This endpoint queries a specific data provider and this data provider has to know what it supposed to return in this endpoint, so the path should come from the data provider.